webapp-kubernetes-typescript/index.ts

```diff
@@ -0,0 +1,96 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as kubernetes from "@pulumi/kubernetes";
+
+// Get some values from the stack configuration, or use defaults
+const config = new pulumi.Config();
+const k8sNamespace = config.get("namespace") || "default";
+const numReplicas = config.getNumber("replicas") || 1;
+const appLabels = {
+    app: "nginx",
+};
+
+// Create a new namespace
+const webServerNs = new kubernetes.core.v1.Namespace("webserver", {metadata: {
+    name: k8sNamespace,
+}});
+
+// Create a new ConfigMap for the Nginx configuration
+const webServerConfig = new kubernetes.core.v1.ConfigMap("webserverconfig", {
+    metadata: {
+        namespace: webServerNs.metadata.apply(m => m.name),
+    },
+    data: {
+        "nginx.conf": `events { }
+http {
+  server {
+    listen 80;
+    root /usr/share/nginx/html;
+    index index.html index.htm index.nginx-debian.html
+    server_name _;
+    location / {
+      try_files $uri $uri/ =404;
+    }
+  }
+}
+`,
+    },
+});
+
+// Create a new Deployment with a user-specified number of replicas
+const webServerDeployment = new kubernetes.apps.v1.Deployment("webserverdeployment", {
+    metadata: {
+        namespace: webServerNs.metadata.apply(m => m.name),
```

Comment:
```suggestion
        namespace: webServerNs.metadata.name,
```
The TS SDK will automatically lift Output values, so apply isn't needed here.

Reply:
Fixed in 0a8a73a